### PR TITLE
Add unit tests DEVAS 1898

### PR DESCRIPTION
**Pull request by paolina92**

### CODE REVIEW

File: assembl/static2/tests/unit/components/debate/common/topPostForm.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  });
+
+  it('should update the subject with the new value', () => {
+    const instance = wrapper.instance();
+    instance.handleSubjectChange({ target: { value: 'Hello' } });
+    const subject = wrapper.state().subject;
+    expect(subject).toEqual('Hello');
+  });
+
+  it('should return a key key to display a message in the alert when the subject is not filled', () => {
```
Comment (pascalhuynh) [inline assembl/static2/tests/unit/components/debate/common/topPostForm.spec.jsx]: type `key key`

File: assembl/static2/tests/unit/components/debate/common/topPostForm.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+};
+
+describe('<TopPostForm /> - with shallow', () => {
+  let wrapper;
+
+  beforeEach(() => {
+    wrapper = shallow(<DumbTopPostForm {...topPostFormProps} />);
+  });
+
+  it('should expand the component', () => {
```
[the remark this answers is not in the document]
Comment (paolina92) [reply assembl/static2/tests/unit/components/debate/common/topPostForm.spec.jsx]: ok I'll change it

[the remark this answers is not in the document]
Comment (paolina92) [reply assembl/static2/tests/unit/components/debate/common/topPostForm.spec.jsx]: oh no. Finally it's not a click and it's more complicated than this... I can show you if you want

File: assembl/static2/js/app/components/debate/common/topPostForm.jsx [hunk 3]
```diff
@@ -146,46 +165,27 @@ class TopPostForm extends React.Component<TopPostFormProps, TopPostFormState> {
...
+      const warningMessage = this.getWarningMessageToDisplay(publicationState, subject, bodyIsEmpty, ideaOnColumn);
+      if (warningMessage) displayAlert('warning', I18n.t(warningMessage));
       this.setState(submittingState(false));
     }
   };
 
   handleInputFocus = promptForLoginOr(() => this.displayForm(true));
 
-  updateBody = (newValue) => {
+  updateBody = (newValue: Object) => {
```
[the remark this answers is not in the document]
Comment (paolina92) [reply assembl/static2/js/app/components/debate/common/topPostForm.jsx]: True but I think that it's not in the scope of this ticket.

**Pull request merged**